src/InstanceSerialization/ClassTransformer.php

```diff
@@ -29,6 +30,10 @@ class ClassTransformer implements CanPack {
    * @throws RuntimeException
    */
   public function pack(Packer $packer, $instance): ?string {
+    if ($instance instanceof Float4Wrapper) {
+      $packer = (new Packer(PackOptions::FORCE_STR | PackOptions::FORCE_FLOAT32))->extendWith(new ClassTransformer());
+      return $packer->pack($instance->value);
```

Comment:
@eugene536 I just released msgpack.php [v0.7.2](https://github.com/rybakit/msgpack.php/releases/tag/v0.7.2) which allows removing this workaround with instantiating a new packer instance and just use `packFloat32()`:
```suggestion
      return $packer->packFloat32($instance->value);
```

Reply:
Thanks for the comment, honestly it wasn't a good name for the wrapper, I've renamed it.
The `packFloat32` is not suitable here as we can pack not only `float` values, but any types (arrays, instances, etc.) forcing floats values to be packed as `float32` inside it.